timbermill-local/src/main/java/com/datorama/oss/timbermill/ElasticsearchClient.java

```diff
@@ -671,62 +678,54 @@ long countByName(String name, String env) throws IOException {
         return countResponse.getCount();
     }
 
-	public void retryFailedRequestsFromMemory() {
-		if (!failedRequests.isEmpty()) {
-			LOG.info("------------------ Failed Requests From Memory Retry Start ------------------");
-			List<Pair<DbBulkRequest, Integer>> list = memoryFailedRequestsAsList();
-			for (Pair<DbBulkRequest, Integer> entry : list) {
-				DbBulkRequest dbBulkRequest = entry.getKey();
-				Integer retryNum = entry.getValue();
-				sendDbBulkRequest(dbBulkRequest, retryNum + 1);
+	boolean handleBulkRequestFailure(DbBulkRequest dbBulkRequest, int tryNum, BulkResponse responses, String failureMessage) {
```

Comment:
method name should describe what it is doing, like shouldRetryBulkrequset,
I would consider taking out the persistence part to have a better separation of the code so it will be easier to understand what each method does.

Reply:
consider adding a timeout for the whole retry process

Reply:
There's a timeout on the request to es

Reply:
also can be private

Reply:
what is the timeout? if it's 60 seconds for example and we retry 3 times and fail every time for the same timeout it's not good..

Reply:
the timeout is 30 seconds but once you send a request you can't cancel on your own before ES timeout is reached.
you can stop waiting for the response of course but it will not be canceled, you will end up writing twice